SYCL/ESIMD/api/functional/type_traits.hpp

```diff
@@ -0,0 +1,38 @@
+//===-- type_traits.hpp - Define functions for iterating with datatypes. --===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+///
+/// \file
+/// This file provides function for iterating with data types.
+///
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <type_traits>
+
+namespace esimd_test {
+namespace api {
+namespace functional {
+namespace type_traits {
+
+template <typename T> constexpr bool is_sycl_type() {
+  return std::is_same_v<T, sycl::half>;
+}
+
+template <typename T> struct is_floating_point {
+  static constexpr bool value{std::is_floating_point_v<T> ||
+                              std::is_same_v<T, sycl::half>};
+};
+
+template <typename T>
+inline constexpr bool is_floating_point_v{is_floating_point<T>::value};
```

Comment:
```suggestion
inline constexpr bool is_sycl_floating_point_v{is_floating_point<T>::value};
```
In case of `is_sycl_floating_point` of course)

Reply:
this comment was applied in this commit: ef6509579513a2452045f0c2cf2c0599a6c3b96b